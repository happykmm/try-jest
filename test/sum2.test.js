jest.mock('../src/sum');

const sum = require('../src/sum');

function LeakingClass() {
}

global.leaks = [];
setInterval(function() {
  for (var i = 0; i < 100; i++) {
    leaks.push(new LeakingClass());
  }
}, 1);

describe('test-suite', () => {  
  test('test-case', () => {
    const data = {
      1: 'a',
      2: 'b',
    };
    sum(15, 'foo', { key: 'value' });
    global.foo = 'foo';
    console.log('global.bar = ', global.bar);
    expect(sum).toMatchSnapshot();
    setTimeout(() => {
      console.log(JSON.stringify(data));
    }, 10000);
  });

  test('adds 1 + 3', (done) => {
    global.bar = 'bar';
    console.log('global.foo = ', global.foo);
    setTimeout(() => {
      console.log(leaks.length);
      done();
    }, 1000);
  });
});