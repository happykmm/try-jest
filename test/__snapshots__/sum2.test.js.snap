// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-suite test-case 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      15,
      "foo",
      Object {
        "key": "value",
      },
    ],
  ],
}
`;
